### PR TITLE
Update Find in Files Package option

### PR DESCRIPTION
**Pull request by melissa-barca**
After some user testing, we determined it makes more sense to separate the `Search in: Package Source` option into two options: `Package Source (/R, /src)` and `Package Tests (/tests)`.

image

This PR also updates the `git grep` command to override any global git configurations the user may have set up on their computer. This could've led to rstudio mis-parsing output received from the command. 
It also removes `.Rd` from the `Search in: R Common Source Files` option based on feedback that the majority of users have these files autogenerated.

### CODE REVIEW

File: src/cpp/session/modules/SessionFind.cpp [hunk 1]
```diff
@@ -1256,6 +1276,11 @@ core::Error runGrepOperation(const GrepOptions& grepOptions, const ReplaceOption
    if (grepOptions.gitFlag())
    {
       cmd = shell_utils::ShellCommand("git");
+      cmd << "-c" << "grep.lineNumber=true";
```
Comment (kevinushey) [inline src/cpp/session/modules/SessionFind.cpp]: Might be worth adding a comment indicating that we set these configuration options to override any potential user-defined options?

File: src/gwt/src/org/rstudio/studio/client/workbench/views/output/find/FindInFilesDialog.ui.xml [hunk 2]
```diff
@@ -45,9 +45,10 @@
          <rw:FormListBox ui:field="listPresetFilePatterns_"
                     styleName="{style.presetFilePatterns}">
             <g:item value="">All Files</g:item>
-            <g:item value="*.r, *.R, *.rnw, *.Rnw, *.rmd, *.Rmd, *.rmarkdown, *.Rmarkdown, *.rhtml, *.Rhtml, *.rd, *.Rd, *.h, *.hpp, *.c, *.cpp">Common R source files (R, C/C++, Rnw, Rmd, Rhtml, Rd)</g:item>
+            <g:item value="*.r, *.R, *.rnw, *.Rnw, *.rmd, *.Rmd, *.rmarkdown, *.Rmarkdown, *.rhtml, *.Rhtml, *.h, *.hpp, *.c, *.cpp">Common R source files (R, C/C++, Rnw, Rmd, Rhtml)</g:item>
             <g:item value="*.r, *.R">R Scripts</g:item>
-            <g:item value="package">Package Source (R/ , src/, tests/)</g:item>
+            <g:item value="packageSource">Package Source (R/ , src/)</g:item>
```
Comment (kevinushey) [inline src/gwt/src/org/rstudio/studio/client/workbench/views/output/find/FindInFilesDialog.ui.xml]: I would prefer "Package Sources" or "Package Source Files" as a label here. (@jmcphers do you have an opinion?)

Comment (jmcphers) [reply src/gwt/src/org/rstudio/studio/client/workbench/views/output/find/FindInFilesDialog.ui.xml]: "Package Sources" (the simpler the better!)

File: src/cpp/session/modules/SessionFind.cpp (hunk 1 above)
Comment (melissa-barca) [reply src/cpp/session/modules/SessionFind.cpp]: Good point, comment added!

File: src/gwt/src/org/rstudio/studio/client/workbench/views/output/find/FindInFilesDialog.ui.xml (hunk 2 above)
Comment (melissa-barca) [reply src/gwt/src/org/rstudio/studio/client/workbench/views/output/find/FindInFilesDialog.ui.xml]: Done!

**Pull request merged**